### DISCUSSION TITLE
net::ERR_CONNECTION_REFUSED When trying to get image from azure blob container using SixLabors.ImageSharp.Web

### DISCUSSION THREAD

**Discussion by dLopes-SE** [SixLabors/ImageSharp.Web#170] [Q&A] [answered]
Hello. I'm trying to use ImageSharp.web for image processing using .net Core. After testing with an image stored in wwwroot, which successfully worked, I'm now trying to process images that are stored in Azure Blob storage and I'm not being successful.

I've followed the Sixlabors tutorials and I can't get an image result on: https://localhost:5001/X/foto-teste10515264.jpg -> X is the container name.

I've got the following configurations in startup.cs:

```csharp
public void ConfigureServices(IServiceCollection services)
{
    services.AddRazorPages();
    services.AddImageSharp()
        .SetRequestParser<QueryCollectionRequestParser>()
        .Configure<AzureBlobStorageImageProviderOptions>(options =>
        {
            // The "BlobContainers" collection allows registration of multiple containers.
            options.BlobContainers.Add(new AzureBlobContainerClientOptions
            {
                ConnectionString = "DefaultEndpointsProtocol=https;AccountName=***;AccountKey=***;EndpointSuffix=core.windows.net",
                ContainerName = "X"
            });
        })
        .AddProvider<AzureBlobStorageImageProvider>()
}
```

And my Configure:

```csharp
public void Configure(IApplicationBuilder app, IWebHostEnvironment env)
{
    if (env.IsDevelopment())
    {
        app.UseDeveloperExceptionPage();
    }
    else
    {
        app.UseExceptionHandler("/Error");
        // The default HSTS value is 30 days. You may want to change this for production scenarios, see https://aka.ms/aspnetcore-hsts.
        app.UseHsts();
    }

    app.UseHttpsRedirection();
    app.UseDefaultFiles();
    app.UseImageSharp();

    app.UseStaticFiles();

    app.UseRouting();

    app.UseAuthorization();

    app.UseEndpoints(endpoints =>
    {
        endpoints.MapRazorPages();
    });
}
```

Thanks in advance.

**Comment by dLopes-SE** [comment, marked as answer]
I solved it by clearing the provides before. Thanks anyway!

**Reply by JimBobSquarePants** [reply]
Yep. providers are iterated in order they are loaded. You need to ensure any providers with increased specificity are checked before the catch all local file provider

**Reply by dLopes-SE** [reply]
@JimBobSquarePants I'm using blob azure storage for caching my images. Do the blobs (used as cache) have a Time-to-live?

**Reply by JimBobSquarePants** [reply]
https://docs.microsoft.com/en-us/azure/cdn/cdn-manage-expiration-of-blob-content#:~:text=You%20can%20choose%20to%20set,to%20general%20web%20delivery%20optimizations.

**Reply by dLopes-SE** [reply]
Thanks @JimBobSquarePants !

**Reply by eddiepantoja** [reply]
How do you clear the provider before?

**Reply by eddiepantoja** [reply]
I figured it out. There is a ClearProviders() method that needs to be chained prior to adding .AddProvider<AzureBlobStorageImageProvider>()
